Make out ops c10-full (with hacky-wrapper)

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49259 Remove `use_c10_dispatcher: full` lines
* #49251 Remove generated_unboxing_wrappers and setManuallyBoxedKernel
* #49220 Remove .impl_UNBOXED() and functionalities associated with it
* #49164 Remove codegen logic to support non-c10-full ops
* #49138 [wip] Making ops c10-full: list of optional tensors
* #49150 Making ops c10-full: ConstQuantizerPtr
* #49146 Making ops c10-full: Storage arguments
* #49088 Making ops c10-full: optional lists
* #49083 Making ops c10 full: optional out arguments
* #49013 Making ops c10-full: Generator arguments
* #49012 Making ops c10 full: out overloads with default arguments
* #49008 Making ops c10-full: Dimname arguments
* #49007 Making ops c10-full: ops blocked by manual registrations
* #49006 Making ops c10 full: ops returning multiple out arguments
* **#48912 Make out ops c10-full (with hacky-wrapper)**

Benchmark:
---
Old (i.e. codegenerated unboxing wrapper + no hacky_wrapper):
```
<torch.utils.benchmark.utils.valgrind_wrapper.timer_interface.CallgrindStats object at 0x7f64d03ebcd0>
torch.absolute(t, out=o)
setup:
  t = torch.empty([1])
  o = torch.empty([1])

                           All          Noisy symbols removed
    Instructions:       657204                     634396
    Baseline:             4192                       3786
100 runs per measurement, 1 thread
```

New (i.e. templated unboxing wrapper + hacky_wrapper):
```
<torch.utils.benchmark.utils.valgrind_wrapper.timer_interface.CallgrindStats object at 0x7fa7de211cd0>
torch.absolute(t, out=o)
setup:
  t = torch.empty([1])
  o = torch.empty([1])

                           All          Noisy symbols removed
    Instructions:       658160                     633996
    Baseline:             4210                       3786
100 runs per measurement, 1 thread
```

Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

smessmer committed:
Make out ops c10-full (with hacky-wrapper)



Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

bhosmer commented:
Windows test failures and XLA build failure look like they might be legit though

bhosmer commented:
Also we should track perf effects - I started a aibench on Phab but probably worth doing one with callgrind as well

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer commented:
Added a benchmark to the PR description

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Benchmark:
---
Old (i.e. codegenerated unboxing wrapper + no hacky_wrapper):
```
<torch.utils.benchmark.utils.valgrind_wrapper.timer_interface.CallgrindStats object at 0x7f64d03ebcd0>
torch.absolute(t, out=o)
setup:
  t = torch.empty([1])
  o = torch.empty([1])

                           All          Noisy symbols removed
    Instructions:       657204                     634396
    Baseline:             4192                       3786
100 runs per measurement, 1 thread
```

New (i.e. templated unboxing wrapper + hacky_wrapper):
```
<torch.utils.benchmark.utils.valgrind_wrapper.timer_interface.CallgrindStats object at 0x7fa7de211cd0>
torch.absolute(t, out=o)
setup:
  t = torch.empty([1])
  o = torch.empty([1])

                           All          Noisy symbols removed
    Instructions:       658160                     633996
    Baseline:             4210                       3786
100 runs per measurement, 1 thread
```

Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Benchmark:
---
Old (i.e. codegenerated unboxing wrapper + no hacky_wrapper):
```
<torch.utils.benchmark.utils.valgrind_wrapper.timer_interface.CallgrindStats object at 0x7f64d03ebcd0>
torch.absolute(t, out=o)
setup:
  t = torch.empty([1])
  o = torch.empty([1])

                           All          Noisy symbols removed
    Instructions:       657204                     634396
    Baseline:             4192                       3786
100 runs per measurement, 1 thread
```

New (i.e. templated unboxing wrapper + hacky_wrapper):
```
<torch.utils.benchmark.utils.valgrind_wrapper.timer_interface.CallgrindStats object at 0x7fa7de211cd0>
torch.absolute(t, out=o)
setup:
  t = torch.empty([1])
  o = torch.empty([1])

                           All          Noisy symbols removed
    Instructions:       658160                     633996
    Baseline:             4210                       3786
100 runs per measurement, 1 thread
```

Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Benchmark:
---
Old (i.e. codegenerated unboxing wrapper + no hacky_wrapper):
```
<torch.utils.benchmark.utils.valgrind_wrapper.timer_interface.CallgrindStats object at 0x7f64d03ebcd0>
torch.absolute(t, out=o)
setup:
  t = torch.empty([1])
  o = torch.empty([1])

                           All          Noisy symbols removed
    Instructions:       657204                     634396
    Baseline:             4192                       3786
100 runs per measurement, 1 thread
```

New (i.e. templated unboxing wrapper + hacky_wrapper):
```
<torch.utils.benchmark.utils.valgrind_wrapper.timer_interface.CallgrindStats object at 0x7fa7de211cd0>
torch.absolute(t, out=o)
setup:
  t = torch.empty([1])
  o = torch.empty([1])

                           All          Noisy symbols removed
    Instructions:       658160                     633996
    Baseline:             4210                       3786
100 runs per measurement, 1 thread
```

Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Benchmark:
---
Old (i.e. codegenerated unboxing wrapper + no hacky_wrapper):
```
<torch.utils.benchmark.utils.valgrind_wrapper.timer_interface.CallgrindStats object at 0x7f64d03ebcd0>
torch.absolute(t, out=o)
setup:
  t = torch.empty([1])
  o = torch.empty([1])

                           All          Noisy symbols removed
    Instructions:       657204                     634396
    Baseline:             4192                       3786
100 runs per measurement, 1 thread
```

New (i.e. templated unboxing wrapper + hacky_wrapper):
```
<torch.utils.benchmark.utils.valgrind_wrapper.timer_interface.CallgrindStats object at 0x7fa7de211cd0>
torch.absolute(t, out=o)
setup:
  t = torch.empty([1])
  o = torch.empty([1])

                           All          Noisy symbols removed
    Instructions:       658160                     633996
    Baseline:             4210                       3786
100 runs per measurement, 1 thread
```

Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]

smessmer committed:
Update on "Make out ops c10-full (with hacky-wrapper)"


Benchmark:
---
Old (i.e. codegenerated unboxing wrapper + no hacky_wrapper):
```
<torch.utils.benchmark.utils.valgrind_wrapper.timer_interface.CallgrindStats object at 0x7f64d03ebcd0>
torch.absolute(t, out=o)
setup:
  t = torch.empty([1])
  o = torch.empty([1])

                           All          Noisy symbols removed
    Instructions:       657204                     634396
    Baseline:             4192                       3786
100 runs per measurement, 1 thread
```

New (i.e. templated unboxing wrapper + hacky_wrapper):
```
<torch.utils.benchmark.utils.valgrind_wrapper.timer_interface.CallgrindStats object at 0x7fa7de211cd0>
torch.absolute(t, out=o)
setup:
  t = torch.empty([1])
  o = torch.empty([1])

                           All          Noisy symbols removed
    Instructions:       658160                     633996
    Baseline:             4210                       3786
100 runs per measurement, 1 thread
```

Differential Revision: [D25363335](https://our.internmc.facebook.com/intern/diff/D25363335/)

[ghstack-poisoned]
